sql/analyzer/resolve_tables.go

```diff
@@ -180,8 +191,85 @@ func handleTableLookupFailure(err error, tableName string, dbName string, a *Ana
 	return nil, err
 }
 
+// unresolveTables is a quick and dirty way to make prepared statement reanalysis
+// resolve the most up-to-date table roots while preserving projections folded into
+// table scans.
+func unresolveTables(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope, sel RuleSelector) (sql.Node, error) {
```

Comment:
This isn't actually safe in the general case, there's a race where the schema of the table changes between prepare and execute

I think this is fine for now, but call it out and add a TODO

We need to detect this edge case and re-analyze the entire plan from scratch if it happens